src/main/scala/zio/kafka/consumer/ConsumerSettings.scala

```diff
@@ -37,6 +37,9 @@ final case class ConsumerSettings(
   def withGroupId(groupId: String): ConsumerSettings =
     withProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId)
 
+  def hasGroupId: Boolean =
+    properties.contains(ConsumerConfig.GROUP_ID_CONFIG)
+
```

Comment:
🟡 i would not expose this method to the lib clients, I would inline where it is called.